### PR TITLE
[docs] update and rework push notifications guide

### LINKED ISSUE #8274: [docs] Make push notifications doc easier to digest
It's pretty long and there is tons of content and code. We should look into how we can make the doc easier to navigate and digest. I think having a table of contents sidebar would be very helpful here.

### PR DESCRIPTION
**Pull request by cruzach**
# Why

The previous version was using the legacy updates module, and was also REALLY long and not so easy to digest. Closes https://github.com/expo/expo/issues/8274

# How

Break apart into these sections:

Screen Shot 2020-06-30 at 5 15 09 PM

each of which is pretty isolated, and takes the user on a journey but now it's a journey with some rest stops included 😄

### CODE REVIEW

**Comment by 1337LutZ** [comment]
Is it an idea to rewrite this example to the newer React Hooks (FC) instead of maintaining old class components? I see a lot of people in the Slack channel "struggling" with the examples since they don't seem to understand you can rewrite class component examples into FC code?

File: docs/pages/push-notifications/receiving-notifications.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  _handleNotificationResponse = response => {
+    /* @info Want to know what the format of the object this listener receives is? Find the specifics in the expo-notifications documentation */
+    this.setState({ response: response });
+    /* @end */
+  };
+
+  render() {
+    return (
+      <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
+        <Text>Origin: {this.state.notification.origin}</Text>
```
Comment (1337LutZ) [inline docs/pages/push-notifications/receiving-notifications.md:64]: This doesn't seem correct, according to the types and your code on: `docs/pages/push-notifications/overview.md` origin doesn't exist anymore and this code doesn't reflect the modifications of the notification object.
It should be or am i missing something?:
```
<View style={{ alignItems: 'center', justifyContent: 'center' }}>
      <Text>Title: {this.state.notification.request.content.title}</Text>
      <Text>Body: {this.state.notification.request.content.body}</Text>
      <Text>Data: {JSON.stringify(this.state.notification.request.content.data)}</Text>
</View>
```

Comment (cruzach) [reply docs/pages/push-notifications/receiving-notifications.md:64]: good catch! i forgot to update the UI for this example

**Comment by cruzach** [comment]
Personally I find a class component a little easier to understand for beginners, that's why I kept the example as a class component

**Comment by 1337LutZ** [comment]
> Personally I find a class component a little easier to understand for beginners, that's why I kept the example as a class component

I totally understand, would be a nice future-feature to maybe have a toggle for both. This would however require more maintenance. However I've only started doing React a few months ago I learned it doing the Hooks way and for me personally class components were much harder to understand since I never got familiar with the "outdated" concept.

Maybe something to consider for the future.

**Comment by cruzach** [comment]
Yeah, great point. Although I wouldn't call class components are "outdated" 😅

**Comment by 1337LutZ** [comment]
@cruzach I just noticed that you did use `componentWillUnmount` in the example, however that has been deprecated and will throw a warning.

https://reactjs.org/blog/2019/08/08/react-v16.9.0.html

**Comment by cruzach** [comment]
@1337LutZ https://github.com/expo/expo/pull/9070

**Pull request merged**